### ISSUE TITLE
Restful API for Add/Update/Remove operations

### ISSUE THREAD

**Issue by Marcnuth** [wagtail/wagtail#4667]
In wagtail's document, it provides some good Restful API for fetch contents via HTTP GET method. See document: http://docs.wagtail.io/en/v2.0/advanced_topics/api/v2/configuration.html

However, when I want to Update/Add/Remove the contents, I cannot find any Restful API to do this. And in its document, it doesn't mention anything about this.

What should I do if I want some Add/Update/Remove Restful API？ and why Wagtail doesn't implement this?

**Comment by gasman** [comment]
Hi @Marcnuth - thanks for raising this. There's a proposed specification for this at https://github.com/wagtail/rfcs/blob/master/text/015-revisions-api-revised.md, but as far as I'm aware there hasn't been any work on implementing it yet.

**Comment by nilesh05apr** [comment]
Hello @gasman is this task done? if not i would love to work on this.

**Comment by gasman** [comment]
@nilesh05apr No, it's not been done yet - you're very welcome to work on it! https://github.com/wagtail/rfcs/blob/main/text/015-revisions-api-revised.md is still the current spec for how we'd like this to work (although the API endpoints for submitting to moderation are probably outdated, since these have since been replaced by the [workflows](https://docs.wagtail.org/en/stable/editor_manual/administrator_tasks/managing_workflows.html) system).

**Comment by NXPY123** [comment]
Are these actions supposed to be added to `PagesAPIViewSet` in `api.v2.views`?

**Comment by thibaudcolas** [comment]
For anyone interested in this, we’ve been working on [RFC 115: Write API](https://github.com/wagtail/rfcs/pull/115) to explore what we have to do to get this added in Wagtail.

**Comment by thibaudcolas** [comment]
This is now implement via #14295, and available in [Wagtail 8.0rc1](https://github.com/wagtail/wagtail/discussions/14501) (feedback welcome!). Will close.

**Comment by thibaudcolas** [comment]
And now in a final release :) Docs: [v3 API index](https://docs.wagtail.org/en/stable/advanced_topics/api/v3/index.html), any feedback on the new API, please head here: [v3 API feedback](https://github.com/wagtail/wagtail/discussions/14531)

For what we’re working on next with the write API, overview: #14565
